### PR TITLE
[nt]  column metric additions

### PR DESCRIPTION
**Pull request by nathaniel-mage**
# Summary

### Improve the column reports page
- Refactor cell and tables for undefined values
- Add progress bar to percentage metrics
- Add more quality metrics to the page.

# Tests

Went on localhost
### Long table
image

### Short Table
image

cc: @johnson-mage 

Note this has big changes to design as it removes the border from the cells for the datatable.

### CODE REVIEW

File: mage_ai/frontend/components/datasets/columns/ColumnReports.tsx [hunk 1]
```diff
@@ -29,10 +29,15 @@ function ColumnReports({
...
   const {
+    average,
     completeness,
     count,
     count_distinct: countDistinct,
     invalid_value_count: invalidValueCount,
+    max,
+    median,
+    min,
+    // mode,
```
Comment (johnson-mage) [inline mage_ai/frontend/components/datasets/columns/ColumnReports.tsx:40]: Is `mode` purposely left out for a reason?

File: mage_ai/frontend/oracle/components/Table/Cell.tsx [hunk 2]
```diff
@@ -117,8 +119,10 @@ function Cell({
         first={cellIndex === 0}
         showBackground={showBackground}
         small={small}
+        vanish={vanish}
       >
-        {cellEl}
+        { !vanish && <>{cellEl} </>
```
Comment (johnson-mage) [inline mage_ai/frontend/oracle/components/Table/Cell.tsx:124]: wrapping `cellEl` in the empty brackets seems unnecessary

**Comment by johnson-mage** [comment]
When there are a lot of values in the Outliers table, the Column Summary table has excessive white space. We want to borders to fit the children content. 
image

**Comment by johnson-mage** [comment]
Average value should be rounded.

**Comment by johnson-mage** [comment]
The second column gets cut off in Column Summary even on full screens. We can change the proportion of the screen that the Warnings/Outliers tables are taking up.

File: mage_ai/frontend/components/datasets/columns/ColumnReports.tsx [hunk 3]
```diff
@@ -48,38 +53,64 @@ function ColumnReports({
...
+      ],
+    },
+    {
+      columnValues: [
+        'Average value', average, [false],
+      ],
+    },
+    // {
+    //   columnValues: [
+    //     'Mode value', mode, [false],
```
Comment (johnson-mage) [inline mage_ai/frontend/components/datasets/columns/ColumnReports.tsx:111]: Same here. Mode commented out?

**Comment by johnson-mage** [comment]
SimpleDataTable looks very similar visually to RowDataTable, and refactoring it might work now to fit what you need it for, but it seems like it's quickly becoming unmaintainable. A better solution seems like it would have been to just use RowDataTable instead. If those comments I left can be fixed quickly, then we can merge it in, but the excessive empty space at the bottom isn't something we'd want in there.

File: mage_ai/frontend/components/datasets/columns/ColumnReports.tsx (hunk 1 above)
Comment (nathaniel-mage) [reply mage_ai/frontend/components/datasets/columns/ColumnReports.tsx:40]: Yeah it didn't work so I thought we could do the metric later. My backends not returning it.

File: mage_ai/frontend/components/datasets/columns/ColumnReports.tsx (hunk 3 above)
Comment (nathaniel-mage) [reply mage_ai/frontend/components/datasets/columns/ColumnReports.tsx:111]: Intentional, see commit comment.

**Review by johnson-mage — APPROVED** [review]
approved after addressing comments. there shudnt be the excess white space in the tables

**Comment by johnson-mage** [comment]
> I couldn't get them to not appear, so I instead will display them as NaN. image
> 
> Also I reverted my border changes as they broke the tables on other pages that encapsulate them inside another component.

@nathaniel-mage Why not "N/A" instead of NaN?

**Comment by johnson-mage** [comment]
> I couldn't get them to not appear, so I instead will display them as NaN. image
> 
> Also I reverted my border changes as they broke the tables on other pages that encapsulate them inside another component.

@nathaniel-mage Not sure if you're addressing this in another PR already, but the bottom corners are also clipped.

**Pull request merged**